homeassistant/components/timer/__init__.py

```diff
@@ -356,9 +357,9 @@ def async_change(self, duration: timedelta) -> None:
         self.async_write_ha_state()
 
     @callback
-    def async_pause(self):
+    def async_pause(self) -> None:
         """Pause a timer."""
-        if self._listener is None:
+        if self._listener is None or self._end is None:
             return
```

Comment:
Why is this code changed?

Reply:
`self._end` needs to be a `datetime`. If not the call to `end.isoformat()` later would fail. In effect, as soon as `self._listener` isn't `None`, `self._end` is also set. So this isn't a functional change but rather one to make mypy happy. I could've also used an `assert` but this seemed better.
https://github.com/home-assistant/core/blob/1d2c23d81c56a38944a77d3537c599136cfa9cc3/homeassistant/components/timer/__init__.py#L396-L404